libs/gi/formula/src/calculator.ts

```diff
@@ -8,19 +8,19 @@ import {
 import type { Member, Read, Sheet, Tag } from './data/util'
 import { reader, tagStr } from './data/util'
 
+const emptyInfo: Info = Object.freeze({ conds: Object.freeze({}) })
 const { arithmetic } = calculation
-const emptyCond: CondInfo = {}
 
-type CondInfo = Partial<
-  Record<
-    Member,
-    Partial<Record<Member, Partial<Record<Sheet, Record<string, number>>>>>
-  >
->
-export type CalcMeta = {
-  tag: Tag | undefined
+type MemRec<V> = Partial<Record<Member, V>>
+type CondInfo = MemRec<MemRec<Partial<Record<Sheet, Record<string, number>>>>>
```

Comment:
Copy the comment from below here as well?
```suggestion
// conds[dst][src][sheet][condName] = condValue
type CondInfo = MemRec<MemRec<Partial<Record<Sheet, Record<string, number>>>>>
```

Reply:
Maybe we should *move* it here